pkg/reconciler/queue_pipelineruns.go

```diff
@@ -34,6 +34,19 @@ func (r *Reconciler) queuePipelineRun(ctx context.Context, logger *zap.SugaredLo
 		return fmt.Errorf("updateError: %w", err)
 	}
 
+	r.secretNS = repo.GetNamespace()
+	// find global repository if set
+	globalRepo, err := r.run.Clients.PipelineAsCode.PipelinesascodeV1alpha1().Repositories(r.run.Info.Kube.Namespace).Get(
+		ctx, r.run.Info.Controller.GlobalRepository, metav1.GetOptions{},
+	)
+	if err == nil && globalRepo != nil {
+		repo.Spec.Merge(globalRepo.Spec)
+	}
+
+	if repo.Spec.GitProvider != nil && repo.Spec.GitProvider.Secret == nil && r.globalRepo.Spec.GitProvider != nil && r.globalRepo.Spec.GitProvider.Secret != nil {
```

Comment:
Shouldn't we also check that the GitProvider on the "local" repo and global repo match? The documentation says "Since the local repository CR has the git_provider.type gitlab like the global repository CR the git provider settings for the GitLab will be taken from the global repository. The secret referenced will be fetched from where the global repository is defined." . The way I read it, they should match to take the secret from the global repo config.

Reply:
@enarha I got answer to your this question now :smile: that we [merge](https://github.com/openshift-pipelines/pipelines-as-code/blob/d56bd5e4bc242719431e41c644ca2b295676f8a5/pkg/apis/pipelinesascode/v1alpha1/types.go#L148) repo with global (you can see above) in that merge it is confirmed that both git providers are equal.